src/dotnet/commands/RestoringCommand.cs

```diff
@@ -20,7 +20,8 @@ private IEnumerable<string> ArgsToForwardToRestore()
                 !a.StartsWith("/t:") &&
                 !a.StartsWith("/target:") &&
                 !a.StartsWith("/ConsoleLoggerParameters:") &&
-                !a.StartsWith("/clp:"));
+                !a.StartsWith("/clp:") &&
+                !a.StartsWith("/p:TargetFramework"));
```

Comment:
It is not designed to catch arbitrarily passed arguments. Just the ones passed by our own parser. Though, at this point, we can't really discern it. So, there is a weird corner case here, but doing it otherwise would come at a higher cost on the code. So, I chose to go this route which will be correct from the functional point of view of implicit restore.